common/components/renderCbs/TransactionFields/GetTransactionFields.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { AppState } from 'reducers';
+import { TypeKeys, FieldAction } from 'actions/transaction';
+
+interface ReduxProps {
+  transactionFields: AppState['transaction']['fields'];
+}
+
+interface IValid {
+  valid: boolean;
+}
+
+type MergeValid<T> = { [P in keyof T]: T[P] & IValid };
+
+interface Props {
+  withFieldValues(
+    values: MergeValid<AppState['transaction']['fields']>
+  ): React.ReactElement<any> | null;
+}
+
+class GetTransactionFieldsClass extends Component<Props & ReduxProps, {}> {
+  public render() {
+    return this.props.withFieldValues(this.getMergedProps());
+  }
+
+  private getMergedProps = (): MergeValid<
+    AppState['transaction']['fields']
+  > => {
+    const { transactionFields } = this.props;
+
+    return Object.keys(transactionFields).reduce((obj, currField: TypeKeys) => {
+      const prop: FieldAction['payload'] = transactionFields[currField];
+      const valid = !!prop.value;
+      const mergedProp = { ...prop, valid };
+      return { ...obj, [currField]: mergedProp };
+    }, {}) as MergeValid<AppState['transaction']['fields']>;
```

Comment:
Not sure if it should be done from the reducer if it can be computed from state. I do use this render cb a lot in other components. I could implement a comparison function or `componentShouldUpdate` to minimize re-renders

Reply:
Or I could split this render cb into a selector and a render cb to seperate the logic from the component